examples/typescript-reactflux/gulpfile.js

```diff
@@ -0,0 +1,332 @@
+  return gulp.src(config.ts_typings, {base: '.'})
+	     .pipe(gulp.dest( config.build ));
+});
+
+gulp.task('src', function() {
+  // You can use multiple globbing patterns as you would with `gulp.src`
+  return gulp.src(config.ts_sources, {base: '.'})
+	     .pipe(gulp.dest( config.build ));
+});
+
+gulp.task('jsx', function() {
+
+  function jsx(all, match)
+  {
+    var reactCode;
+
+    if( match )
+    {
+      try {
+	       reactCode = reactTools.transform(match, { harmony: false });
+      }
+      catch (ex) {
+	       gutil.error('Problem transforming the following:\n' + match + '\n\n' + ex);
+      }
+    }
+
+    // In case you need to debug this:
+    // gutil.log('jsx =>' + match);
+
+    return '(' + reactCode + ')';
+  }
+
+  return gulp.src([config.ts_sources,'!src/react/ReactJSX.ts'])
+	    .pipe(replace(/ReactJSX.*`([^`\\]*(\\.[^`\\]*)*)`([^`\\\)]*(\\.[^`\\\)]*)*)\)/gm, jsx))
+	    .pipe(replace(/import ReactJSX.*$/gm, ''))
+	    // .pipe(log())
+	    .pipe(gulp.dest(config.build + '/src/'));
+});
+
+gulp.task('ts', function() {
+  // Minify and copy all JavaScript (except vendor scripts)
+  // with sourcemaps all the way down
+
+  var ts_options = {			      // Use to override the default options, http://gruntjs.com/configuring-tasks#options
+      target: 'es5',			      // 'es3' (default) | 'es5'
+      sourceMap: false,			      // true (default) | false
+      declaration: false,		      // true | false (default)
+      removeComments: false,		      // true (default) | false
+      noImplicitAny: true,		      // Warn on expressions and declarations with an implied 'any' type.
+      module: 'commonjs',
+      outDir: config.js_generated,
+      fast: 'never',
+      typescript: require('typescript')
+  };
+
+  return gulp.src( config.ts_jsx)
+	.pipe(ts(ts_options))
+	.pipe(log())
+	.pipe(gulp.dest( config.build + '/js/'));
+});
+
+gulp.task('bundle', function() {
+  // Minify and copy all JavaScript (except vendor scripts)
+  // with sourcemaps all the way down
+
+  // 'function ieKeyFix(key)' => 'var ieKeyFix = function(key)'
+  // node_modules/store/store.js declares a function 'ieKeyFix' in the middle of another function and triggers:
+  // ERROR - functions can only be declared at top level or immediately within another function in ES5 strict mode
+
+  // 'this.module !== module' => 'window.module !== module' 
+  // node_modules/store/store.js has some questionable code that triggered:
+  // Uncaught TypeError: Cannot read property 'module' of undefined
+  
+  return gulp.src( config.build + '/js/app.jsx.js')
+	.pipe(browserify({
+	  insertGlobals : true,
+	  debug: true
+	}))
+  .pipe(replace(/function ieKeyFix\(key\)/gm, 'var ieKeyFix = function(key)'))
+  .pipe(replace(/this.module !== module/gm, 'window.module !== module'))
+	.pipe(concat('bundle.js'))
+	.pipe(log())
+	.pipe(gulp.dest( config.web + '/js/'));
+});
+
+gulp.task('minify', function() {
```

Comment:
though I understand how a build step might be necessary in this example, minification seems out of scope.